File: 14. Longest Common Prefix/solution.py
#   In fact, this is not a solution, but it is so elegant and elegant that I want to share it.

#   Runtime: 36 ms, faster than 93.87% of Python3 online submissions for Longest Common Prefix.
#   Memory Usage: 13.2 MB, less than 52.95% of Python3 online submissions for Longest Common Prefix.

class Solution:
    def longestCommonPrefix(self, strs):
        Zip, res = zip(*strs), ''
        for ch in Zip:
            if len(set(ch)) > 1: break
            res += ch[0]
        return res

File: 14. Longest Common Prefix/solution.cpp
/*
    Runtime: 4 ms, faster than 98.91% of C++ online submissions for Longest Common Prefix.
    Memory Usage: 9 MB, less than 60.84% of C++ online submissions for Longest Common Prefix.
*/

class Solution {
public:
    string longestCommonPrefix(const vector<string>& v) {
        if (v.size() == 0) return "";
        size_t len_str = v[0].size();
        for(size_t i = 1; i < v.size(); ++i){
        	size_t tmp = 0;
        	for(size_t j = 0; j < min(len_str, v[i].size()); ++j){
        		if(v[0][j] != v[i][j])
        			break;
        		else 
        			++tmp;
        	}
        	len_str = min(len_str, tmp);
        }
        return v[0].substr(0, len_str);
    }
};
